src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -410,3 +401,27 @@ func enableBorda(bordaReportInterval time.Duration, bordaSamplePercentage float6
 
 	ops.RegisterReporter(reporter)
 }
+
+func includeInSample(deviceID string, samplePercentage float64) bool {
+	if samplePercentage == 0 {
+		return false
+	}
+
+	// Sample a subset of device IDs.
+	// DeviceID is expected to be a Base64 encoded 48-bit (6 byte) MAC address
+	deviceIDBytes, base64Err := base64.StdEncoding.DecodeString(deviceID)
```

Comment:
I'll add some unit test coverage in #5099 since I had to touch the tests there anyway.

Regarding DeviceID - I use it (as encoded) because it's available on both the client and server and allows us to correlate.  For borda, this affects sampling because we want sampling on the client and server to be consistent so that for those device IDs included in the sample, we have end-to-end coverage.  I like keeping it consistent with Loggly to because this ensures that we have full coverage across Loggly and Borda for whichever device IDs are in the sample.